### DISCUSSION TITLE
av.audio.frame.to_ndarray, min and max value ?

### DISCUSSION THREAD

**Discussion by robinechuca** [PyAV-Org/PyAV#1073] [1. Help]
Hello, I asked the question on this forum: https://www.developpez.net/forums/d2143965/autres-langages/python/bibliotheques-tierces/pyav-av-audioframe-np_ndarray-valeurs-min-max-intervalle-type/
Their answer is to contact you directly !

In the case of floating formats, the vast majority of values are between -1 and 1. But not always! Sometimes they are a bit higher.
How far can it go? How to make sure to have values between -1 and 1?
If you know anything about it, it's all good!

Thanks,
Robin

**Comment by phineas-pta** [comment]
`.to_ndarray()` accepts arg `format`, for audio stream, list of accepted values is in https://github.com/PyAV-Org/PyAV/blob/main/av/audio/frame.pyx

search google for « PCM sample format » to understand what they mean

the default format should be integers, but if you use float, values are between -1 and 1

reference 1:https://wiki.multimedia.cx/index.php/PCM

reference 2: https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/samplefmt.h (read the comment before the line `enum AVSampleFormat`)

**Reply by robinechuca** [reply]
Thanks you very much for this accurate answer!
